rct229/rules/section5/section5rule28.py

```diff
@@ -0,0 +1,139 @@
+            matched_proposed_surfaces = match_lists_by_id(
+                baseline_surfaces, proposed_surfaces
+            )
+
+            proposed_baseline_surface_pairs = zip(
+                baseline_surfaces, matched_proposed_surfaces
+            )
+
+            return [
+                UserBaselineProposedVals(None, surface_b, surface_p)
+                for surface_b, surface_p in proposed_baseline_surface_pairs
+                if scc[surface_b["id"]] == SCC.UNREGULATED
+            ]
+
+        class UnregulatedSurfaceRule(RuleDefinitionListIndexedBase):
+            def __init__(self):
+                super(
+                    Section5Rule28.BuildingRule.UnregulatedSurfaceRule, self
+                ).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, True),
+                    list_path="subsurfaces[*]"
+                )
+
+            class UnregulatedSubsurfaceRule(RuleDefinitionBase):
+                def __init__(self):
+                    super(
+                        Section5Rule28.BuildingRule.UnregulatedSurfaceRule.UnregulatedSubsurfaceRule,
+                        self,
+                    ).__init__(
+                        rmrs_used=UserBaselineProposedVals(False, True, True),
+                        required_fields={
+                            "$": [
+                                "u_factor",
+                                "solar_heat_gain_coefficient",
+                                "glazed_area",
+                                "opaque_area",
+                            ]
+                        },
+                    )
+
+                def get_calc_vals(self, context, data=None):
+                    subsurface_b = context.baseline
+                    subsurface_p = context.proposed
+
+                    return {
+                        "baseline_subsurface_u_factor": subsurface_b["u_factor"],
+                        "proposed_subsurface_u_factor": subsurface_p["u_factor"],
+                        "baseline_subsurface_shgc": subsurface_b[
+                            "solar_heat_gain_coefficient"
+                        ],
+                        "proposed_subsurface_shgc": subsurface_p[
+                            "solar_heat_gain_coefficient"
+                        ],
+                        "baseline_subsurface_glazed_area": subsurface_b["glazed_area"],
+                        "proposed_subsurface_glazed_area": subsurface_p["glazed_area"],
+                        "baseline_subsurface_opaque_area": subsurface_b["opaque_area"],
+                        "proposed_subsurface_opaque_area": subsurface_p["opaque_area"],
+                    }
+
+                def rule_check(self, context, calc_vals, data=None):
+                    return std_equal(
+                            calc_vals["baseline_subsurface_u_factor"],
+                            calc_vals["proposed_subsurface_u_factor"],
+                        ) and std_equal(
```

Comment:
The rest of these comparisons are not correct. They reference calc_vals that do not exist.